src/bin/build.rs

```diff
@@ -90,10 +93,12 @@ pub fn execute(options: Options, config: &Config) -> CliResult {
 
     let root = find_root_manifest_for_wd(options.flag_manifest_path, config.cwd())?;
 
-    let spec = if options.flag_all {
-        Packages::All
-    } else {
-        Packages::Packages(&options.flag_package)
+    let spec = match (options.flag_all, &options.flag_exclude) {
+        (true, exclude) if exclude.is_empty() => Packages::All,
+        (true, exclude) => Packages::OptOut(exclude),
+        (false, exclude) if !exclude.is_empty() => panic!("--exclude can only be used together \
```

Comment:
Yep, `bail!` is for this use case: https://github.com/rust-lang/cargo/blob/13d92c64d0153d95dbabeb49b828bbbef4b1bb34/src/cargo/core/shell.rs#L136-L145